src/libraries/System.Data.OleDb/src/ArchitectureSpecificHelpers.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Data.OleDb
+{
+    internal interface ItagDBPROPINFO
+    {
+        int dwPropertyID { get; }
+        int dwFlags { get; }
+        int vtType { get; }
+        object vValue { get; }
+        string pwszDescription { get; }
+    }
+
+    internal interface ItagDBPROP
+    {
+        OleDbPropertyStatus dwStatus { get; }
+        object vValue { get; }
+        int dwPropertyID { get; }
+    }
+
+    internal sealed partial class tagDBPROPINFO_x86 : ItagDBPROPINFO
+    {
+        int ItagDBPROPINFO.dwPropertyID => this.dwPropertyID;
+
+        int ItagDBPROPINFO.dwFlags => this.dwFlags;
+
+        int ItagDBPROPINFO.vtType => this.vtType;
+
+        object ItagDBPROPINFO.vValue => this.vValue;
+
+        string ItagDBPROPINFO.pwszDescription => this.pwszDescription;
+    }
+
+    internal sealed partial class tagDBPROPINFO : ItagDBPROPINFO
+    {
+        int ItagDBPROPINFO.dwPropertyID => this.dwPropertyID;
+
+        int ItagDBPROPINFO.dwFlags => this.dwFlags;
+
+        int ItagDBPROPINFO.vtType => this.vtType;
+
+        object ItagDBPROPINFO.vValue => this.vValue;
+
+        string ItagDBPROPINFO.pwszDescription => this.pwszDescription;
+
+    }
+
+    internal sealed partial class tagDBPROP_x86 : ItagDBPROP
+    {
+        OleDbPropertyStatus ItagDBPROP.dwStatus => this.dwStatus;
+
+        object ItagDBPROP.vValue => this.vValue;
+
+        int ItagDBPROP.dwPropertyID => this.dwPropertyID;
+    }
+
+    internal sealed partial class tagDBPROP : ItagDBPROP
+    {
+        OleDbPropertyStatus ItagDBPROP.dwStatus => this.dwStatus;
+
+        object ItagDBPROP.vValue => this.vValue;
+
+        int ItagDBPROP.dwPropertyID => this.dwPropertyID;
+    }
+
+    internal static class ArchitectureSpecificHelpers
+    {
+        internal static ItagDBPROPINFO CreateTagDbPropInfo()
```

Comment:
These static methods can be moved into `OleDb_Util.cs` or consider moving them to OleDbStruct.cs itself.
That is the location where such utility methods are placed.